iroh/src/magicsock/relay_actor.rs

```diff
@@ -600,19 +592,20 @@ impl RelayActor {
             }
         }
         let url = found_relay.as_ref().unwrap_or(url);
-        self.active_relay_handle(url).await
+        self.active_relay_handle(url)
     }
 
     /// Returns the handle of the [`ActiveRelayActor`].
-    async fn active_relay_handle(&mut self, url: &RelayUrl) -> &ActiveRelayHandle {
+    fn active_relay_handle(&mut self, url: &RelayUrl) -> &ActiveRelayHandle {
         if !self.active_relays.contains_key(url) {
```

Comment:
this could probably be written nicer and more efficiently using the `.entry()` API

Reply:
Problem is that `self.start_active_relay()` still takes `&mut self` because it does `self.active_relay_tasks.spawn()` and that `spawn` needs a mutable reference to the `JoinSet`.  Even moving that JoinSet to an argument of `start_active_relay` won't help.

Reply:
I guess you could put the `JoinSet` in an `Arc<Mutex<JoinSet<_>>>` and that mutex would be un-contended anyway because this is an actor and we get the joy of `&mut self`-exclusivity.  But that is a lot of kludge and moving complexity to the actor struct itself.  Here at least the kludge stays local.